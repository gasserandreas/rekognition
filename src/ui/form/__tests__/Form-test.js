/* global React, shallow, mount, render, renderer, testSnapshot */

import {
  Field,
  StyledFieldLabel,
  StyledFieldFeedback,
  TextInput,
  StyledTextInput,
  StyledCheckBox,
  StyledFieldRow,
} from '../Form';

const getShallowField = (props = {}) => shallow(<Field {...props} />);
const getMountField = (props = {}) => mount(<Field {...props} />);

// Field component tests
it('Field should render correctly', () => {
  testSnapshot(<Field />);
});

it('Extended Field should render correctly', () => {
  const Input = <TextInput value="" onChange={() => {}} />;
  const label = 'Custom label';
  testSnapshot(
    <Field
      label={label}
    >{Input}</Field>);
});

it('Field should render content', () => {
  const props = {
    children: <TextInput value="" onChange={() => {}} />,
  };
  const output = getMountField(props);
  expect(output.props().children).toEqual(props.children);
});

it('Field should not render label if no label is passed in', () => {
  const output = getShallowField();
  expect(output.find(StyledFieldLabel).length).toEqual(0);
});

it('Field should render label with custom props', () => {
  const props = {
    id: 'id12-3456',
    error: 'custom error',
    label: 'label',
  };
  const output = getShallowField(props);

  // should contains label content
  expect(output.contains(props.label)).toEqual(true);

  // check for label component
  const label = output.find(StyledFieldLabel);
  expect(label);

  // check for label props
  const labelProps = label.props();
  
  expect(labelProps.children).toEqual(props.label);
  expect(labelProps.htmlFor).toEqual(props.id);
});

it('Field should not render Feedback if no error is passed in', () => {
  const output = getShallowField({ error: '' });
  expect(output.find(StyledFieldFeedback).length).toEqual(0);
});

it('Field should render Feedback with error', () => {
  const props = {
    error: 'error',
  };
  const output = getShallowField(props);
  
  const feedback = output.find(StyledFieldFeedback);
  expect(feedback);

  const feedbackProps = feedback.props();
  expect(feedbackProps.children).toEqual(props.error);
});

// Input tests
it('Input styles should be consistent', () => {
  testSnapshot(<StyledTextInput />);
});

it('Disabled input should render correctly', () => {
  testSnapshot(<StyledTextInput disabled/>);
});

it('Error Input should render correctly', () => {
  testSnapshot(<StyledTextInput error="error" />);
});

// Checkbox tests
it('Checkbox styles should be consistent', () => {
  testSnapshot(<StyledCheckBox />);
});

it('Checkbox Input should render correctly', () => {
  testSnapshot(<StyledCheckBox error="error" />);
});

// Test StyledFieldRow
it('FieldRow styles should be consistent', () => {
  testSnapshot(<StyledFieldRow />);
});
