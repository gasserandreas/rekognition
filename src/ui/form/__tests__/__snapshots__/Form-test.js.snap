// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Input should render correctly 1`] = `
<label
  className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 bVmNDQ"
  onClick={[Function]}
>
  <div
    className="StyledCheckBox-sc-1dbk5ju-6 eRXCj StyledBox-sc-13pk1d4-0 emCRMK"
  >
    <input
      className="sc-bwzfXH bmqJpw StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 eGHaUo"
      onBlur={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <div
      className="StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 fxKnpz StyledBox-sc-13pk1d4-0 jFlQDV"
    />
  </div>
</label>
`;

exports[`Checkbox styles should be consistent 1`] = `
<label
  className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 bVmNDQ"
  onClick={[Function]}
>
  <div
    className="StyledCheckBox-sc-1dbk5ju-6 eRXCj StyledBox-sc-13pk1d4-0 emCRMK"
  >
    <input
      className="sc-bwzfXH bGgvvP StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 eGHaUo"
      onBlur={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
    <div
      className="StyledCheckBox__StyledCheckBoxBox-sc-1dbk5ju-3 fxKnpz StyledBox-sc-13pk1d4-0 jFlQDV"
    />
  </div>
</label>
`;

exports[`Disabled input should render correctly 1`] = `
<div
  className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
>
  <input
    autoComplete="off"
    className="sc-bdVaJa bjqEvM StyledTextInput-sc-1x30a0s-0 iQGivQ"
    disabled={true}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`Error Input should render correctly 1`] = `
<div
  className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
>
  <input
    autoComplete="off"
    className="sc-bdVaJa fDbrzy StyledTextInput-sc-1x30a0s-0 iQGivQ"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;

exports[`Extended Field should render correctly 1`] = `
<div
  className="sc-ifAKCX fXmiTd"
>
  <label
    className="sc-bxivhb kgFgTv"
  >
    Custom label
  </label>
  <div
    className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
  >
    <input
      autoComplete="off"
      className="sc-bdVaJa bjqEvM StyledTextInput-sc-1x30a0s-0 iQGivQ"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      value=""
    />
  </div>
</div>
`;

exports[`Field should render correctly 1`] = `
<div
  className="sc-ifAKCX fXmiTd"
/>
`;

exports[`FieldRow styles should be consistent 1`] = `
<div
  className="sc-EHOje jzFSNI StyledBox-sc-13pk1d4-0 IRSNj"
/>
`;

exports[`Input styles should be consistent 1`] = `
<div
  className="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 yIdXy"
>
  <input
    autoComplete="off"
    className="sc-bdVaJa bjqEvM StyledTextInput-sc-1x30a0s-0 iQGivQ"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
  />
</div>
`;
